cpp/src/arrow/util/byte_stream_split.h

```diff
@@ -163,21 +171,421 @@ void ByteStreamSplitEncodeSSE2(const uint8_t* raw_values, const size_t num_value
         final_result[i * 2 + 1] = _mm_unpackhi_epi64(tmp[i], tmp[i + 2]);
       }
     }
-    for (size_t i = 0; i < num_streams; ++i) {
+    for (size_t i = 0; i < kNumStreams; ++i) {
       _mm_storeu_si128(&output_buffer_streams[i][block_index], final_result[i]);
     }
   }
 }
+#endif  // ARROW_HAVE_SSE4_2
+
+#if defined(ARROW_HAVE_AVX2)
+template <typename T>
+void ByteStreamSplitDecodeAVX2(const uint8_t* data, int64_t num_values, int64_t stride,
```

Comment:
sorry for the churn, I think lets change these back to Avx/Simd, etc (see my reply to @pitrou about this being more consistent with the style guide).

Reply:
Thanks. Changed to camel cased per the coding guide.